src/libCrypto/Schnorr.cpp

```diff
@@ -120,6 +125,8 @@ void BIGNUMSerialize::SetNumber(vector<unsigned char> & dst, unsigned int offset
 
 shared_ptr<EC_POINT> ECPOINTSerialize::GetNumber(const vector<unsigned char> & src, unsigned int offset, unsigned int size)
 {
+    lock_guard<mutex> g(m_mutexECPOINT);
+
     shared_ptr<BIGNUM> bnvalue = BIGNUMSerialize::GetNumber(src, offset, size);
```

Comment:
Is this function locking two locks? because it calls GetNumber which also aquires a lock?